HLTrigger/Configuration/python/customizeHLTforPatatrack.py

```diff
@@ -471,6 +470,112 @@ def customise_gpu_ecal(process):
+    process.hcalConvertedEffectivePedestalWidthsGPUESProducer.label1 = "withTopoEff"
+    process.load("RecoLocalCalo.HcalRecProducers.hcalQIECodersGPUESProducer_cfi")
+    process.load("RecoLocalCalo.HcalRecProducers.hcalRecoParamsWithPulseShapesGPUESProducer_cfi")
+    process.load("RecoLocalCalo.HcalRecProducers.hcalRespCorrsGPUESProducer_cfi")
+    process.load("RecoLocalCalo.HcalRecProducers.hcalTimeCorrsGPUESProducer_cfi")
+    process.load("RecoLocalCalo.HcalRecProducers.hcalQIETypesGPUESProducer_cfi")
+    process.load("RecoLocalCalo.HcalRecProducers.hcalSiPMParametersGPUESProducer_cfi")
+    process.load("RecoLocalCalo.HcalRecProducers.hcalSiPMCharacteristicsGPUESProducer_cfi")
+
+
+    # Modules and EDAliases
+
+    # hltHfprereco and hltHoreco need the HBHE digis in CPU format, which is not
+    # available if the unpacker is run on the gpu.
+    # So, run the legacy unacker on the cpu, then convert the HBHE digis in SoA
+    # format and copthem to the gpu.
+    #
+    #from EventFilter.HcalRawToDigi.hcalRawToDigiGPU_cfi import hcalRawToDigiGPU as _hcalRawToDigiGPU
+    #process.hltHcalDigisGPU = _hcalRawToDigiGPU.clone(
+    #    digisLabelF5HB = "",
+    #    digisLabelF01HE = "",
+    #    digisLabelF3HB = "",
+    #)
+    #
+    #process.hltHcalDigis = cms.EDProducer("HcalCPUDigisProducer",
+    #    digisLabelF01HEIn = cms.InputTag("hltHcalDigisGPU"),
+    #    digisLabelF5HBIn = cms.InputTag("hltHcalDigisGPU"),
+    #    digisLabelF3HBIn = cms.InputTag("hltHcalDigisGPU"),
+    #    digisLabelF01HEOut = cms.string(""),
+    #    digisLabelF5HBOut = cms.string(""),
+    #    digisLabelF3HBOut = cms.string("")
+    #)
+
+    process.hltHcalDigisGPU = cms.EDProducer("HcalDigisProducerGPU",
+        hbheDigisLabel = cms.InputTag("hltHcalDigis"),
+        qie11DigiLabel = cms.InputTag("hltHcalDigis"),
+        digisLabelF01HE = cms.string(""),
+        digisLabelF5HB = cms.string(""),
+        digisLabelF3HB = cms.string(""),
+        maxChannelsF01HE = cms.uint32(10000),
+        maxChannelsF5HB = cms.uint32(10000),
+        maxChannelsF3HB = cms.uint32(10000),
+        nsamplesF01HE = cms.uint32(8),
+        nsamplesF5HB = cms.uint32(8),
+        nsamplesF3HB = cms.uint32(8)
+    )
+
+    from RecoLocalCalo.HcalRecProducers.hbheRecHitProducerGPU_cfi import hbheRecHitProducerGPU as _hbheRecHitProducerGPU
+    process.hltHbherecoGPU = _hbheRecHitProducerGPU.clone(
+        digisLabelF01HE = "hltHcalDigisGPU",
+        digisLabelF5HB = "hltHcalDigisGPU",
+        digisLabelF3HB = "hltHcalDigisGPU",
+        recHitsLabelM0HBHE = ""
+    )
+
+    from RecoLocalCalo.HcalRecProducers.hcalCPURecHitsProducer_cfi import hcalCPURecHitsProducer as _hcalCPURecHitsProducer
+    process.hltHbhereco = _hcalCPURecHitsProducer.clone(
+        recHitsM0LabelIn = "hltHbherecoGPU",
+        recHitsM0LabelOut = "",
+        recHitsLegacyLabelOut = ""
+    )
+
+
+    # Sequences
+
+    process.HLTDoLocalHcalSequence = cms.Sequence(
```

Comment:
regarding hf/ho in there, do they run at hlt or not? I was told that ho/hf reco are not part of the sequence at all.
gpu unpacker does not save hf/ho, it can be added, but right now it does not.

Reply:
Yes, we do unpack and use HF and HO at HLT.
Will stay with the CPU unpacker for the time being.